libraries/standard/mqtt/src/iot_mqtt_validate.c

```diff
@@ -93,10 +108,11 @@ bool _IotMqtt_ValidateConnect( const IotMqttConnectInfo_t * pConnectInfo )
     {
         if( pConnectInfo->pPreviousSubscriptions != NULL )
         {
-            if( pConnectInfo->previousSubscriptionCount == 0 )
+            if( _IotMqtt_ValidateSubscriptionList( IOT_MQTT_SUBSCRIBE,
```

Comment:
The subscription count is already validated in [`_IotMqtt_ValidateSubscriptionList()`](https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/v4_beta/libraries/standard/mqtt/src/iot_mqtt_validate.c#L344).